src/actions/participants.js

```diff
@@ -12,15 +12,21 @@ const actions = createActions(
 const setRandomParticipant = () => (dispatch, getState) => {
 
     const participantList = getState().participants.listOfParticipants;
-    const selectedParticipant = participantList[random(0, participantList.length - 1)];
+    const index = random(0, participantList.length - 1);
+    const selectedParticipant = { ...participantList[index], index };
 
     dispatch(actions.setSelectedParticipant({ selectedParticipant }));
 };
 
 const removeParticipant = index => (dispatch, getState) => {
 
-    const participantList = getState().participants.listOfParticipants;
-    const listOfParticipants = participantList.filter(participant => participant.index !== index);
+    const listOfParticipants = getState().participants.listOfParticipants.slice();
+
+    if (index === 0) {
+        listOfParticipants.shift();
+    } else {
+        listOfParticipants.splice(index);
```

Comment:
@filgaia when using splice we need to pass the `deleteCount` number so it delete exactly one number of the array and not from the index and beyond. In this case it should be `.splice(index, 1)`
See: https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Array/splice#Par%C3%A1metros